core/modules/user/user.module

```diff
@@ -1772,6 +1790,106 @@ function user_page_title($account) {
   return is_object($account) ? user_format_name($account) : '';
 }
 
+/**
+ * Implements hook_init().
+ *
+ * Suppresses the use of layouts and blocks on the user login, register, and
+ * reset pages if configured to use the simple login page.
+ */
+function user_init() {
+  if (user_page_is_simplified_page()) {
+    layout_suppress(TRUE);
+  }
+}
+
+/**
+ * Implements hook_custom_theme().
+ *
+ * Activates the admin theme for login, register, and reset pages if
+ * configured to use the simple login page. Note this by-passes the
+ * "view the administration theme" permission to show the admin theme to
+ * anonymous users.
+ *
+ * @see user_init()
+ */
+function user_custom_theme() {
+  if (user_page_is_simplified_page()) {
+    return config_get('system.core', 'admin_theme');
+  }
+}
+
+/**
+ * Implements hook_preprocess_page().
+ */
+function user_preprocess_page(&$variables) {
+  if (user_is_logged_in()) {
+    return;
+  }
+  if (user_page_is_simplified_page()) {
+    $variables['classes'][] = 'user-simplified-page-body';
+  }
+}
+
+/**
+ * Implements hook_preprocess_page_component().
+ * 
+ * Suppresses tabs when user form appearance is "links".
+ */
+function user_preprocess_page_components(&$variables) {
+  if (user_is_logged_in()) {
+    return;
+  }
+  // Suppress tabs if for user forms when using "links" appearance.
+  if (in_array(current_path(), array('user/login', 'user/register', 'user/password')) &&  user_login_appearance_setting() == 'links') {
```

Comment:
I'd find it more readable, if the array of three user/* pages would be outside the if clause.

Actually exactly the same check is done below ...
```
in_array(current_path(), array('user/login', 'user/password', 'user/register'))
```
So, why not making it a callback? But even just pulling it out of the if clause seems an improvement for readability. :wink: Whatever you consider the best solution.